src/Layer.js

```diff
@@ -281,22 +291,57 @@ export default class Layer {
         this._fireUpdateOnNextRender = true;
     }
 
+    /**
+     *  Check if the layer has any dataframe.
+     *
+     * @memberof carto.Layer
+     * @instance
+     * @api
+     */
     hasDataframes () {
         return this._renderLayer.hasDataframes();
     }
 
+    /**
+     *  Get layer id
+     *
+     * @memberof carto.Layer
+     * @instance
```

Comment:
Documenting all of this and exposing it on the API has consequences, some of this methods are probably broken, they certainly are untested, users haven't requested them and some of them can be confusing. For example, `getSource` and `getViz` semantics are not clear, when an `update` is performed the source is not changed immediately. All of this should be documented if we decide to put it on the API. To be honest, I think documenting this is not a good idea at this moment